architecture/protocol.go

```diff
@@ -45,78 +44,70 @@ const MAX_JOB_SIZE int64 = 65536 // 2^16
 type Command struct {
 	Name           CommandName
 	RawCommand     string
-	Params         map[string]string
 	WaitingForMore bool
 	MoreToSend     bool
 	Err            error
 	Job            Job
+
+	Tube     string
```

Comment:
Let me think about this.  `command.Params` would still have to be public since it's needed in `operation`.